app/controllers/display_ad_events_controller.rb

```diff
@@ -15,27 +15,22 @@ def create
   private
 
   def update_display_ads_data
-    return if skip_ad_update?
-
-    @display_ad = DisplayAd.find(display_ad_event_params[:display_ad_id])
-
-    num_impressions = @display_ad.display_ad_events.where(category: "impression").size
-    num_clicks = @display_ad.display_ad_events.where(category: "click").size
-    rate = num_clicks.to_f / num_impressions
-
-    @display_ad.update_columns(
-      success_rate: rate,
-      clicks_count: num_clicks,
-      impressions_count: num_impressions,
-    )
+    ThrottledCall.perform(:display_ads_data_update, throttle_for: 15.minutes) do
```

Comment:
15 minutes is pretty arbitrary. In #12779 we replaced a `rand(6)` call with `5.minutes` so I picked 15 for this `rand(20)` call. If you prefer another value let me know and I update it.